services/accounts/accounts.go

```diff
@@ -97,6 +103,50 @@ func (api *API) AddAccountWithMnemonic(
 	return api.SaveAccounts(ctx, []accounts.Account{account})
 }
 
+func (api *API) AddAccountWithMnemonicAndPath(
```

Comment:
Can u update the addAccountWithMnemonic to call this function?
That way we avoid code duplication

Reply:
sure. Will create a new func for that